### DISCUSSION TITLE
Gyroscope/Touchpad compatibility to DS4 emulation from Moonlight client

### DISCUSSION THREAD

**Discussion by ld892012** [LizardByte/community#184] [Sunshine]
Some games needs gyroscope or touchpad to work properly like emulated games of some recent games but only axes and buttons is sent to host by Moonlight.

Sunshine gives possibility to emulate DS4 controller instead of X360 but no motion/touchscreen data is transferred.

At this moment, there is no way to send motion data from client to host except for Dolphin or Cemu by create DSU server on client but it not work on all platforms and don't work outside of LAN.

The combo Moonlight/Sunshine is already a great alternative (selectable screen, multi client, custom resolution/fps) and it can be even more greater with this kind of features. 👍

**Comment by ld892012** [comment]
Now Moonlight can support gyro and touchpad in this [Commit](https://github.com/moonlight-stream/moonlight-qt/commit/9af58af5e4aa536083ad62997135c1f9c03430ec) ! 👍

**Comment by cgutman** [comment]
Closing as a duplicate of LizardByte/Sunshine#1200
